### PR TITLE
Add pyrodigal-gv

### PR DESCRIPTION
**Pull request by apcamargo**
This PR adds [`pyrodigal-gv`](https://github.com/althonos/pyrodigal-gv) by @althonos.

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by althonos** [comment]
You beat me to it 🙃

**Comment by apcamargo** [comment]
@BiocondaBot please update

**Comment by apcamargo** [comment]
@althonos any idea what might be causing the test to hang?

**Comment by althonos** [comment]
I'm not sure, I cannot even get the logs!

There's a chance that's the generation of the `.pyc` file which can be requiring loads of memory?

**Comment by gbouras13** [comment]
The result looks similar to this error https://github.com/bioconda/bioconda-recipes/issues/41480, but not sure if this helps resolve it.

**Comment by mencian** [comment]
I am running into same issue locally:
```
$ bioconda-utils build --packages pyrodigal-gv
[1]    2183 killed     bioconda-utils build --packages pyrodigal-gv
```

**Comment by apcamargo** [comment]
@mencian did you get a log when running it locally?

**Comment by mencian** [comment]
@apcamargo unfortunately no, that was all that was output

**Comment by gbouras13** [comment]
I just locally ran

```
bioconda-utils build --packages pyrodigal-gv bioconda-recipes/recipes/pyrodigal-gv
```

This was the stderr:

```
## Package Plan ##

  environment location: /Users/a1667917/mambaforge/envs/bioconda-utils/conda-bld/pyrodigal-gv_1695349953897/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold


The following NEW packages will be INSTALLED:

    archspec:        0.2.1-pyhd8ed1ab_0         conda-forge
    bzip2:           1.0.8-h0d85af4_4           conda-forge
    ca-certificates: 2023.7.22-h8857fd0_0       conda-forge
    click:           8.1.7-unix_pyh707e725_0    conda-forge
    libffi:          3.4.2-h0d85af4_5           conda-forge
    libsqlite:       3.43.0-h58db7d2_0          conda-forge
    libzlib:         1.2.13-h8a1eda9_5          conda-forge
    ncurses:         6.4-hf0c8a7f_0             conda-forge
    openssl:         3.1.3-h8a1eda9_0           conda-forge
    pip:             23.2.1-pyhd8ed1ab_0        conda-forge
    pyrodigal:       3.0.0-py310he07d9a1_0      bioconda
    pyrodigal-gv:    0.1.0-pyh7cba7a3_0         local
    python:          3.10.12-had23ca6_0_cpython conda-forge
    python_abi:      3.10-4_cp310               conda-forge
    readline:        8.2-h9e318b2_1             conda-forge
    setuptools:      68.2.2-pyhd8ed1ab_0        conda-forge
    six:             1.16.0-pyh6c4a22f_0        conda-forge
    tk:              8.6.12-h5dbffcc_0          conda-forge
    tzdata:          2023c-h71feb2d_0           conda-forge
    wheel:           0.41.2-pyhd8ed1ab_0        conda-forge
    xz:              5.2.6-h775f41a_0           conda-forge

Preparing transaction: ...working... done
Verifying transaction: ...working... done
Executing transaction: ...working... done
Traceback (most recent call last):
  File "/Users/a1667917/mambaforge/envs/bioconda-utils/conda-bld/pyrodigal-gv_1695349953897/test_tmp/run_test.py", line 2, in <module>
    import pyrodigal_gv
  File "/Users/a1667917/mambaforge/envs/bioconda-utils/conda-bld/pyrodigal-gv_1695349953897/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold/lib/python3.10/site-packages/pyrodigal_gv/__init__.py", line 11, in <module>
    from .meta import METAGENOMIC_BINS, METAGENOMIC_BINS_VIRAL
ModuleNotFoundError: No module named 'pyrodigal_gv.meta'
Tests failed for pyrodigal-gv-0.1.0-pyh7cba7a3_0.tar.bz2 - moving package to /Users/a1667917/mambaforge/envs/bioconda-utils/conda-bld/broken
WARNING:conda_build.build:Tests failed for pyrodigal-gv-0.1.0-pyh7cba7a3_0.tar.bz2 - moving package to /Users/a1667917/mambaforge/envs/bioconda-utils/conda-bld/broken
export PREFIX=/Users/a1667917/mambaforge/envs/bioconda-utils/conda-bld/pyrodigal-gv_1695349953897/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold
export SRC_DIR=/Users/a1667917/mambaforge/envs/bioconda-utils/conda-bld/pyrodigal-gv_1695349953897/test_tmp
import: 'pyrodigal_gv'
TESTS FAILED: pyrodigal-gv-0.1.0-pyh7cba7a3_0.tar.bz2

12:07:59 BIOCONDA ERROR BUILD FAILED recipes/pyrodigal-gv
12:07:59 BIOCONDA INFO (COMMAND) conda build purge
12:08:02 BIOCONDA ERROR BUILD SUMMARY: of 1 recipes, 1 failed and 0 were skipped. Details of recipes and environments follow.
12:08:02 BIOCONDA ERROR BUILD SUMMARY: FAILED recipe recipes/pyrodigal-gv
```

**Comment by rpetit3** [comment]
I got this error as well, which suggests a few tweaks to the pip install might be needed

**Comment by althonos** [comment]
Maybe it would be worth using the [wheel file](https://docs.conda.io/projects/conda-build/en/stable/user-guide/wheel-files.html#building-a-conda-package-from-a-wheel-file) directly to build the package?

**Comment by apcamargo** [comment]
That could work, but I'm not sure how Bioconda would handle this, since wheels are Python version-specific. @rpetit3, do you know of any precedent for this?

**Comment by gbouras13** [comment]
So my hack fix didn't work but it does fix building the package locally for me. Not especially surprising of course :)

**Comment by apcamargo** [comment]
Thank for the try, @gbouras13.

So maybe that error is not what is causing the test to hang? Maybe we could try a non-import test. Not sure if that would work either because importing is unavoidable.

**Comment by althonos** [comment]
There is a proper test set included btw, you could use `python -m unittest pyrodigal_gv.tests` for the tests (like in the `pyrodigal` recipe).

**Comment by apcamargo** [comment]
Still hanging indefinitely. Maybe there's a way to get the log of the test?

**Comment by gbouras13** [comment]
Based on the similar issues I linked #41480, I’d say it’s most likely that some step (possibly importing the meta models) is breaking the container’s memory or thread limits

**Comment by apcamargo** [comment]
@BiocondaBot please update

**Comment by apcamargo** [comment]
@gbouras13 this recipe doesn't compile anything and there doesn't seem to be any step that would require a lot of memory. I'm not really sure what the problem is. :/

@bioconda/python any ideas of what might be going on here.

**Comment by althonos** [comment]
@apcamargo : I removed all weird hacks from `v0.2.0` so hopefully after you update the recipe this builds like a normal Python package and doesn't cause the weird stalling :+1:

**Comment by althonos** [comment]
Whoop whoop :partying_face: :partying_face: :partying_face: Thanks for the update @gbouras13 !

**Comment by gbouras13** [comment]
You’re the legend @althonos. I’ll leave it to antonio to test it some more locally if he wants and merge it in tomorrow or when he’s happy :)

**Comment by apcamargo** [comment]
@BiocondaBot please add label

File: recipes/pyrodigal-gv/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+test:
+  imports:
+    - pyrodigal_gv
+  commands:
+    - python -m unittest -vv pyrodigal_gv.tests
+
+about:
+  home: https://github.com/althonos/pyrodigal-gv
+  summary: A Pyrodigal extension to predict genes in giant viruses and viruses with alternative genetic code.
+  license: GPL-3.0
```
Comment (mencian) [inline recipes/pyrodigal-gv/meta.yaml:41]:
```suggestion
  license: GPL-3.0-or-later
  license_family: GPL3
```

**Review by mencian — COMMENTED** [review]
A minor suggestion to use a SPDX identifier for the license field. Otherwise, looks good to me!

**Review by mencian — APPROVED** [review]
LGTM! Thanks everyone!

**Pull request merged**